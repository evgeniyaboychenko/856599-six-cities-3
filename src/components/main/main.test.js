import React from 'react';
import renderer from 'react-test-renderer';
import Main from './main.jsx';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {MemoryRouter} from "react-router-dom";

const mockStore = configureStore([]);
jest.mock(`../map/map.jsx`);

const OFFER_IMAGES = [`room.jpg`, `apartment-01.jpg`];
const OFFER_NAMES = [`Canal View Prinsengracht`, `Nice, cozy, warm big bed apartment`];
const OFFER_TYPES = [`apartment`, `privet room`];
const COORDINATES = [[52.3909553943508, 4.85309666406198], [52.369553943508, 4.85309666406198], [52.3909553943508, 4.929309666406198], [52.3809553943508, 4.939309666406198]];

const CITIES = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];
const COORDINATES_CITY = [[48.856663, 2.351556], [50.930779, 6.938399], [50.854283, 4.352131], [52.373057, 4.892557], [53.552645, 9.966287], [51.230569, 6.787428]];

const activeCity = {
  id: `1`,
  name: CITIES[0],
  coordinatesCity: COORDINATES_CITY[0],
};

const cities = [{
  id: `1`,
  name: CITIES[0],
  coordinatesCity: COORDINATES_CITY[0],
},
{
  id: `2`,
  name: CITIES[1],
  coordinatesCity: COORDINATES_CITY[1],
},
{
  id: `3`,
  name: CITIES[2],
  coordinatesCity: COORDINATES_CITY[2],
}
];

const offerCards = [
  {
    id: `1`,
    coordinates: COORDINATES[0],
    photos: OFFER_IMAGES,
    name: OFFER_NAMES[0],
    rating: 2,
    price: 100,
    type: OFFER_TYPES[0],
    isPremium: true,
  },
  {
    id: `2`,
    coordinates: COORDINATES[1],
    photos: OFFER_IMAGES,
    name: OFFER_NAMES[1],
    rating: 3,
    price: 120,
    type: OFFER_TYPES[1],
    isPremium: false,
  }];

const offersNear = [
  {
    id: `11`,
    photos: OFFER_IMAGES,
    name: OFFER_NAMES[0],
    rating: 2,
    price: 100,
    type: OFFER_TYPES[0],
    isPremium: true,
  },
  {
    id: `22`,
    photos: OFFER_IMAGES,
    name: OFFER_NAMES[1],
    rating: 3,
    price: 120,
    type: OFFER_TYPES[1],
    isPremium: false,
  },
  {
    id: `23`,
    photos: OFFER_IMAGES,
    name: OFFER_NAMES[1],
    rating: 3,
    price: 120,
    
    type: OFFER_TYPES[1],
    isPremium: false,
  }
];

const offerCardsEmpty = [];

it(`should Main render correctly`, () => {
  const store = mockStore({
    city: activeCity,
    offers: offerCards,
    activeSortItem: `Popular`,
    offersNear
  });
  const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Main
            offersCount = {6}
            activeCity = {activeCity}
            cities = {cities}
          />
        </MemoryRouter>
      </Provider>).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`should Main render correctly if offers-count equal null`, () => {
  const store = mockStore({
    city: activeCity,
    offers: offerCardsEmpty,
    activeSortItem: `Popular`,
    offersNear
  });
  const tree = renderer.create(
      <Provider store={store}>
        <Main
          offersCount = {0}
          activeCity = {activeCity}
          cities = {cities}
        />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
