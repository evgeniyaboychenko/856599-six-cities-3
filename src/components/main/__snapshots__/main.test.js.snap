// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Main render correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                  style={
                    Object {
                      "backgroundImage": "url(https://htmlacademy-react-3.appspot.com/six-citiesnike.jpg)",
                    }
                  }
                />
                <span
                  className="header__user-name user__name"
                >
                  nike@n.ru
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="#"
            >
              <span>
                Paris
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
            >
              <span>
                Cologne
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
            >
              <span>
                Brussels
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            3
             places to stay in 
            Paris
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              onClick={[Function]}
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  href="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom"
            >
              <li
                className="places__option places__option--active"
                onClick={[Function]}
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="cities__places-list places__list tabs__content"
          />
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          />
        </div>
      </div>
    </div>
    3
  </main>
</div>
`;

exports[`should Main render correctly if offers-count equal null 1`] = `
<div
  style={
    Object {
      "content": "",
      "marginLeft": "50%",
      "marginTop": "200px",
      "position": "absolute",
    }
  }
>
  <svg
    height="64px"
    version="1.0"
    viewBox="0 0 128 128"
    width="64px"
    xmlSpace="preserve"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <g>
      <circle
        cx="16"
        cy="64"
        fill="#000000"
        fillOpacity="1"
        r="16"
      />
      <circle
        cx="16"
        cy="64"
        fill="#000000"
        fillOpacity="1"
        r="14.344"
        transform="rotate(45 64 64)"
      />
      <circle
        cx="16"
        cy="64"
        fill="#000000"
        fillOpacity="1"
        r="12.531"
        transform="rotate(90 64 64)"
      />
      <circle
        cx="16"
        cy="64"
        fill="#000000"
        fillOpacity="1"
        r="10.75"
        transform="rotate(135 64 64)"
      />
      <circle
        cx="16"
        cy="64"
        fill="#000000"
        fillOpacity="1"
        r="10.063"
        transform="rotate(180 64 64)"
      />
      <circle
        cx="16"
        cy="64"
        fill="#000000"
        fillOpacity="1"
        r="8.063"
        transform="rotate(225 64 64)"
      />
      <circle
        cx="16"
        cy="64"
        fill="#000000"
        fillOpacity="1"
        r="6.438"
        transform="rotate(270 64 64)"
      />
      <circle
        cx="16"
        cy="64"
        fill="#000000"
        fillOpacity="1"
        r="5.375"
        transform="rotate(315 64 64)"
      />
      <animateTransform
        attributeName="transform"
        calcMode="discrete"
        dur="720ms"
        repeatCount="indefinite"
        type="rotate"
        values="0 64 64;315 64 64;270 64 64;225 64 64;180 64 64;135 64 64;90 64 64;45 64 64"
      />
    </g>
  </svg>
</div>
`;
